src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -409,22 +444,29 @@ private void EmitOpCode(OpCode opCode, MethodBase mBase)
             else
             {
                 var cInfo = mBase as ConstructorInfo;
-                if(cInfo != null) ilGen.Emit(opCode, cInfo);
+                if (cInfo != null) ilGen.Emit(opCode, cInfo);
             }
             writer.WriteLine($"{opCode} {mBase}");
         }
 
         private void EmitOpCode(OpCode opCode, double val)
         {
+            if (!EmitEnabled) return;
             ilGen.Emit(opCode, val);
             writer.WriteLine($"{opCode} {val}");
         }
 
         private void EmitOpCode(OpCode opCode, long val)
         {
+            if (!EmitEnabled) return;
```

Comment:
Do we really need this additional flag or can we just say `if(compilePass = CompilePass.GatherTypeInfo) return;`?

Reply:
will look into simplifying it